src/zimscraperlib/zim/creator.py

```diff
@@ -122,14 +113,86 @@ def start(self):
         if getattr(self, "main_path", None):
             self.set_mainpath(self.main_path)
 
-        if getattr(self, "metadata", None):
-            self.update_metadata(**self.metadata)
+        if not self.__all_metadata_is_set:
+            raise ValueError("Mandatory metadata are not all set.")
+        self.__add_metadata(**self.metadata)
+
         return self
 
-    def update_metadata(self, **kwargs):
+    def __add_metadata(self, **kwargs):
         if kwargs:
             for name, value in kwargs.items():
-                self.add_metadata(name, value)
+                if value:
+                    self.add_metadata(name, value)
+
+    def config_metadata(
+        self,
+        *,
+        Name: str,
+        Language: str = "eng",
+        Title: str,
+        Description: str,
+        LongDescription: Optional[str] = None,
+        Creator: str,
+        Publisher: str,
+        Date: Union[datetime.datetime, datetime.date, str] = datetime.datetime.now(),
+        Tags: Optional[Union[Iterable[str], str]] = None,
+        Scraper: Optional[str] = None,
+        Flavour: Optional[str] = None,
+        Source: Optional[str] = None,
+        License: Optional[str] = None,
+        Relation: Optional[str] = None,
+        **extras: str,
+    ) -> Creator:
+        """
+        A chaining functions which configures the metadata of the Creator class.
+        You must set all mandatory metadata in this phase.
+
+        Parameters:
+            check out: https://wiki.openzim.org/wiki/Metadata
+
+        Returns:
+            Creator
+        """
+        self.metadata.update(
+            {
+                "Name": Name,
+                "Title": Title,
+                "Creator": Creator,
+                "Publisher": Publisher,
+                "Date": Date,
+                "Description": Description,
+                "Language": Language,
+                "License": License,
+                "LongDescription": LongDescription,
+                "Tags": Tags,
+                "Relation": Relation,
+                "Flavour": Flavour,
+                "Source": Source,
+                "Scraper": Scraper,
+            }
+        )
+        self.metadata.update(extras)
+        language = self.metadata.get("Language", "").split(",")
+        self.config_indexing(True, language[0])
```

Comment:
I'm a little worried about this part of the change.  let `config_indexing` is alway true, is that a good idea?

Reply:
It is. It's important to understand that pylibzim is *just a wrapper* around libzim. In that it copies the API.
scraperlib instead is a tool for scraper writers. There we make opinionated choices to make writing scrapers easier… without removing functionnalty.

In a scraper, 99% of the time, you want indexing, so we default to it.
One can still call `config_indexing(False, "")` afterwards should indexing not be wanted